src/Control/Eff/Cut.hs

```diff
@@ -60,8 +60,7 @@ cutfalse = throwError CutFalse
 -- but not non-determinism
 call :: forall a r. Member Choose r => Eff (Exc CutFalse ': r) a -> Eff r a
 call m = loop [] m where
-  loop :: Member Choose r
-       => [Eff (Exc CutFalse ': r) a]
+  loop :: [Eff (Exc CutFalse ': r) a]
```

Comment:
@power-fungus please remove these changes from this PR

Reply:
See the older travis-builds. ghc-8.0.1 issues a warning about the extra constraint and since `Werror` is enabled, the whole build turns red. Therefore, I would leave it in or adapt the build process (mute error or drop ghc-8.0.1) until this gets fixed.

Reply:
@power-fungus could we tackle adding `ghc-8.0.1` in another PR? once we've got the mechanical thing of using `stack` instead of `nix` sorted then let's look into those issues

Reply:
got it

Reply:
@suhailshergill should be fixed now